File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/repositories/ReactiveReadOnlyRepository.java
package io.github.sanyavertolet.edukate.backend.repositories;

import org.springframework.data.repository.NoRepositoryBean;
import org.springframework.data.repository.Repository;
import reactor.core.publisher.Mono;

@NoRepositoryBean
public interface ReactiveReadOnlyRepository<T, ID> extends Repository<T, ID> {
    Mono<T> findById(ID id);
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/repositories/ReactiveReadOnlyRepository.kt
package io.github.sanyavertolet.edukate.backend.repositories

import org.springframework.data.repository.NoRepositoryBean
import org.springframework.data.repository.Repository
import reactor.core.publisher.Mono

@NoRepositoryBean
interface ReactiveReadOnlyRepository<T : Any, ID : Any> : Repository<T, ID> {
    fun findById(id: ID): Mono<T>
}
